### ISSUE TITLE
`IllegalImport` does not warn about static imports

### ISSUE THREAD

**Issue by Stephan202** [checkstyle/checkstyle#4954]
* Given:
    ```xml
    <module name="IllegalImport">
        <property name="illegalClasses" value="org\.testng\.AssertJUnit\.*" />
        <property name="regexp" value="true" />
    </module>
    ```
* When:
    ```java
    import org.testng.AssertJUnit;
    ```
* Then:
    ```
    [INFO] --- maven-checkstyle-plugin:2.17:check (run-checkstyle) @ fulfillment-vrs ---
    [INFO] There is 1 error reported by Checkstyle 8.1 with /some/path/target/checkstyle-rules.xml ruleset.
    [ERROR] some/path/Test.java:[8,1] (imports) IllegalImport: Illegal import - org.testng.AssertJUnit.
    ```
* But when
    ```java
    import static org.testng.AssertJUnit.assertEquals;
    ```
* Then no violation is reported.

Can `IllegalImport` be extended to support static imports, or should I use some other kind of check?

**Comment by Stephan202** [comment]
NB: the non-regex config below has exactly the same behavior:
```xml
<module name="IllegalImport">
    <property name="illegalClasses" value="org.testng.AssertJUnit" />
</module>
```

**Comment by rnveach** [comment]
@Stephan202 
> `org\.testng\.AssertJUnit\.*`

Your configuration's regular expression is wrong.

Here is a working example:
````
$ cat TestClass.java
import static org.testng.AssertJUnit.assertEquals;

public class TestClass {
    void method() {
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="IllegalImport">
    <property name="illegalClasses" value="org\.testng\.AssertJUnit\..*" />
    <property name="regexp" value="true" />
</module>
    </module>
</module>

$ java -jar checkstyle-8.1-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:1:1: Illegal import - org.testng.AssertJUnit.assertEquals. [IllegalImport]
Audit done.
Checkstyle ends with 1 errors.
````

> the non-regex config below has exactly the same behavior

Without the regular expression you must set the entire import for `illegalClasses`.
If you used `illegalPkgs`, you wouldn't need the method name.

@romani For static imports, why don't we drop off the last identifier which is a method or field name?
It makes the configuration look slightly weird and we have no examples in our documentation for static imports for this check.

**Comment by Stephan202** [comment]
@rnveach thanks! Looked at this for so long and still missed it. (I guess because this is actually just one line in a rather long list, I went `\.`-blind.)

> @romani For static imports, why don't we drop off the last identifier which is a method or field name?
It makes the configuration look slightly weird and we have no examples in our documentation for static imports for this check.

I like this idea. Right now `illegalClasses` basically only does the "expected thing" w.r.t. static imports if one employs regexes.

**Comment by markkolich** [comment]
@Stephan202 @rnveach @romani I just stubbed my toe on this as well:

```xml
<module name="IllegalImport">
  <!-- Use java.nio.charset.StandardCharsets instead. -->
  <property name="illegalClasses" value="com.google.common.base.Charsets"/>
</module>
```

This fails:
```java
import com.google.common.base.Charsets;
```

This does not:
```java
import static com.google.common.base.Charsets.UTF_8;
```

Using `regexp=true` solved it but I agree with the consensus here that the resulting configuration looks a bit kludgy:

```xml
<property name="illegalClasses" value="com\.google\.common\.base\.Charsets.*"/>
<property name="regexp" value="true" />
```

> @romani For static imports, why don't we drop off the last identifier which is a method or field name?
It makes the configuration look slightly weird and we have no examples in our documentation for static imports for this check.

+1

**Comment by Michael1993** [comment]
What if you want to enforce somethings to be static imported? E.g.: `org.testng.Assert.assertEquals` should be static imported instead of writing `Assert.assertEquals`.

With the proposed change this would be impossible, no? I think the current implementation provides greater flexibility.
